src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -49,6 +52,28 @@ public class ArduinoBoard : Board.Board, IDisposable
 
         private ILogger _logger;
 
+        static ArduinoBoard()
+        {
+            // We add all known modes to the list, even though we don't really support them all in the core
+            _knownSupportedModes.Add(SupportedMode.DigitalInput);
+            _knownSupportedModes.Add(SupportedMode.DigitalOutput);
+            _knownSupportedModes.Add(SupportedMode.AnalogInput);
+            _knownSupportedModes.Add(SupportedMode.Pwm);
+            _knownSupportedModes.Add(SupportedMode.Servo);
```

Comment:
`_knownSupportedModes` could be a ReadOnlyList if you used a `List<SupportedMode>` in a local here and then constructed `ReadOnlyList` from that local

Reply:
ah.. just noticed this is extensible